move/axelar_gateway/sources/gateway.move

```diff
@@ -174,6 +175,19 @@ entry fun rotate_signers(
     )
 }
 
+/// This function should only be called once
+/// (checks should be made on versioned to ensure this)
+/// It upgrades the version control to the new version control.
+entry fun migrate(self: &mut Gateway) {
+    let (v0, cap) = self.inner.remove_value_for_upgrade<Gateway_v0>();
+    let v1 = v0.migrate(version_control());
```

Comment:
make the migration more interesting. e.g. add some message approval, and maybe another field to the gateway object, and add a check post upgrade?

Reply:
also from the doc,

For the second upgrade, also include a change to a dependency contract, such as Utils or so. The deployment scripts should handle upgrading the dependency as well on a change, instead of reusing the existing config address. Can we also include upgrading the example contract to point to the new gateway contract?

Reply:
what is "some message approval"? Messages can be approved already. Can you be more specific? What check post upgrade?

Reply:
since tests for allowed functions cannot be actually automated, I would have to write a custom function that lives on deployments that does nothing for our actual gateway to be able to test this 'new' field, how do you propose we go about this?